Control the policy controller monitoring resources from chart, enhanc…

senanz opened the pull request:
…ment for the current implmentation to hadd all the resources by default, The new change add avail to pass resourcesNames through the chart with list of resources comma sperataed for which resources to be monitored by the policy controller, the default is all resources if the flag wasn't presented in the chart

This PR resolves https://github.com/sigstore/policy-controller/issues/1388 

Signed-off-by: Senan Zedan (EXT-Nokia) <senan.zedan.ext@nokia.com>

#### Summary
 The new change add avail to pass resourcesNames through the chart with list of resources comma sperataed for which resources to be monitored by the policy controller, the default is all resources if the flag wasn't presented in the chart

codysoyland committed:
Add check for ACR registry in ACR credential helper (#1658)

Signed-off-by: Cody Soyland <codysoyland@github.com>
Signed-off-by: Senan Zedan (EXT-Nokia) <senan.zedan.ext@nokia.com>

dependabot[bot] committed:
chore(deps): Bump google-github-actions/auth from 2.1.6 to 2.1.7 (#1683)

Bumps [google-github-actions/auth](https://github.com/google-github-actions/auth) from 2.1.6 to 2.1.7.
- [Release notes](https://github.com/google-github-actions/auth/releases)
- [Changelog](https://github.com/google-github-actions/auth/blob/main/CHANGELOG.md)
- [Commits](https://github.com/google-github-actions/auth/compare/8254fb75a33b976a221574d287e93919e6a36f70...6fc4af4b145ae7821d527454aa9bd537d1f2dc5f)

---
updated-dependencies:
- dependency-name: google-github-actions/auth
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Signed-off-by: Senan Zedan (EXT-Nokia) <senan.zedan.ext@nokia.com>

Commit:
Control the policy controller monitoring resources from chart, enhancment for the current implmentation to hadd all the resources by default, The new change add avail to pass resourcesNames through the chart with list of resources comma sperataed for which resources to be monitored by the policy controller, the default is all resources if the flag wasn't presented in the chart

Signed-off-by: Senan Zedan (EXT-Nokia) <senan.zedan.ext@nokia.com>

Commit:
Control the policy controller monitoring resources from chart, enhancment for the current implmentation to hadd all the resources by default, The new change add avail to pass resourcesNames through the chart with list of resources comma sperataed for which resources to be monitored by the policy controller, the default is all resources if the flag wasn't presented in the chart

Signed-off-by: Senan Zedan (EXT-Nokia) <senan.zedan.ext@nokia.com>

senanz commented:
@vaikas - Could you please look into this?

vaikas commented:
Could you add some tests, maybe here: https://github.com/sigstore/policy-controller/tree/main/test

senanz committed:
Merge branch 'sigstore:main' into issues_1388

senanz committed:
Merge branch 'sigstore:main' into issues_1388

Commit:
Merge branch 'issues_1388' of https://github.com/senanz/policy-controller into issues_1388

Commit:
Merge branch 'issues_1388' of https://github.com/senanz/policy-controller into issues_1388

Commit:
Merge branch 'issues_1388' of https://github.com/senanz/policy-controller into issues_1388

Commit:
Merge branch 'issues_1388' of https://github.com/senanz/policy-controller into issues_1388

vaikas commented:
> @vaikas - per your request, test added.

That does not really test any of this new code. You'd have to add a test that launches policy controller with these new flags, here's one example where we change the policy-controller behaviour by the flags:
https://github.com/sigstore/policy-controller/blob/b3af2f053ba9b652c96a99b08580e75ed2788cbc/.github/workflows/kind-cluster-image-policy-tsa.yaml#L125

So, create a new kustomize file that launches policy controller with say --resource-name=pods, and customize it like here:
https://github.com/sigstore/policy-controller/blob/b3af2f053ba9b652c96a99b08580e75ed2788cbc/.github/workflows/kind-cluster-image-policy-tsa.yaml#L136

And then after the policy-controller has been started with the flags under test, you'd run your new test:
https://github.com/sigstore/policy-controller/blob/b3af2f053ba9b652c96a99b08580e75ed2788cbc/.github/workflows/kind-cluster-image-policy-tsa.yaml#L177

It should at the bare minimum have a negative test and a positive test. So, if you're using resource-name pods, then launching a deployment with a failing config should succeed if you instead launch a pod.

senanz committed:
Merge branch 'sigstore:main' into issues_1388

senanz committed:
Merge branch 'sigstore:main' into issues_1388

senanz committed:
Merge branch 'sigstore:main' into issues_1388

Commit:
Merge branch 'issues_1388' of https://github.com/senanz/policy-controller into issues_1388

vaikas commented on .github/workflows/kind-cluster-custom-resources.yaml:
```diff
@@ -0,0 +1,180 @@
+
+jobs:
+  cip-test-trustroot-bring-your-own-keys:
+    name: ClusterImagePolicy e2e tests with TrustRoot - Bring Your Own Keys
```
Nit: Change the test name, and the job name to match your new functionality.

vaikas commented on .github/workflows/kind-cluster-custom-resources.yaml:
```diff
@@ -0,0 +1,180 @@
+      - name: Install cluster + sigstore
+        uses: sigstore/scaffolding/actions/setup@main
+        env:
+          INSTALL_TSA: true
```
I don't think you necessarily need this, but 🤷

senanz replied:
Acknolwedged, let's have it now for the simplicty.

vaikas commented on .github/workflows/kind-cluster-custom-resources.yaml:
```diff
@@ -0,0 +1,180 @@
+          kubectl rollout status --timeout 5m --namespace cosign-system deployments/webhook
+          echo "TUF_ROOT_FILE=./root.json" >> $GITHUB_ENV
+
+      - name: Checkout TSA for testing.
```
And you probably don't need this either here since we don't test TSA.

senanz replied:
Removed.

vaikas commented on .github/workflows/kind-cluster-custom-resources.yaml:
```diff
@@ -0,0 +1,180 @@
+            exit -1
+          fi
+
+      - name: Run Custom resources Tests
```
You can remove L141 to here, since this is the only thing we're really testing here. No need to test the TSA again.

senanz replied:
You are right, removed.

vaikas commented on test/e2e_test_policy_custom_resource.sh:
```diff
@@ -0,0 +1,68 @@
+fi
+echo '::endgroup::'
+
+echo '::group:: Cleanup'
```
I'd add two more test cases, something like:
`kubectl create deployment ...` with container image that has a tag
Then `kubectl get deployment` and check that the container image in the spec has not been modified, and then `kubectl get pod` and check that the container image in the spec has been modified.

vaikas commented on test/testdata/policy-controller/e2e/test-deployment-with-custom-resource.yaml:
```diff
@@ -0,0 +1,36 @@
+          - args:
+              - resource-name="pods"
+          - name: demo-custom-resource
+          image: TEST_IMAGE
```
The problem here is that the TEST_IMAGE as created has a full digest in it so it's harder to tell if it's been resolved to digest or not. I'd suggest that you have an image that has a tag, which would without this change be resolved to digest for a deployment, so creating a deployment with a tag here would be the simplest way to verify the functionality works. Then the resulting pods should have the digests resolved.

senanz replied:
Test was added.

vaikas commented on test/testdata/policy-controller/e2e/test-deployment-with-custom-resource.yaml:
```diff
@@ -0,0 +1,36 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  namespace: demo-scalable-custom-resource
```
nit: use a different namespace, say: test-deployment-custom-resource

senanz replied:
Thanks, updated.

vaikas reviewed (changes requested):
Looking great! There's just some nits for naming, namespaces. One change that should be made it so make sure the tag->digest resolution does not actually happen. The webhook configuration is a good check, but it's easy enough to also make sure the actual resource (deployment) does not get modified.

senanz committed:
Merge branch 'sigstore:main' into issues_1388

vaikas commented:
Looks like some of the changes I requested, and you marked as fixed have not made it in to this PR 🤔 ?

Also, looks like bunch of `sed` commands are failing, for example:
https://github.com/sigstore/policy-controller/actions/runs/12234409121/job/34140387122?pr=1687#step:14:132

senanz commented:
Hi Ville,
Fixed them locally and working in the last comment to add tests.
Updated PR will raise soon.

On Mon, 9 Dec 2024 at 18:48 Ville Aikas ***@***.***> wrote:

> Looks like some of the changes I requested, and you marked as fixed have
> not made it in to this PR 🤔 ?
>
> Also, looks like bunch of sed commands are failing, for example:
>
> https://github.com/sigstore/policy-controller/actions/runs/12234409121/job/34140387122?pr=1687#step:14:132
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/sigstore/policy-controller/pull/1687#issuecomment-2528676880>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIDIUC3MMEW4UY6HLKCMUCL2EXCUBAVCNFSM6AAAAABRCWYFPSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKMRYGY3TMOBYGA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

senanz committed:
Merge branch 'sigstore:main' into issues_1388

Commit:
--amend

Commit:
--amend

Commit:
Merge branch 'issues_1388' of https://github.com/senanz/policy-controller into issues_1388

Commit:
Merge branch 'issues_1388' of https://github.com/senanz/policy-controller into issues_1388

senanz committed:
Merge branch 'sigstore:main' into issues_1388

vaikas commented on cmd/webhook/main.go:
```diff
@@ -78,6 +79,13 @@ var (
 	disableTUF = flag.Bool("disable-tuf", false, "Disable TUF support.")
 
+	// Validate specific resources.
+    // https://github.com/sigstore/policy-controller/issues/1388
```
nit, but the format looks off here?

senanz replied:
seems something related to git.
I'm using intelji and format looks good.
![image](https://github.com/user-attachments/assets/0f4d7976-1cf1-44bc-8494-ecb3f6da46a9)

senanz replied:
Ran the command: gofmt -s -w .

vaikas commented on cmd/webhook/main.go:
```diff
@@ -116,6 +124,8 @@ func main() {
 
 	flag.Parse()
+	resourcesNamesList := strings.Split(*resourcesNames, ",")
+    listResources = append(listResources, resourcesNamesList...)
```
and here. Maybe run gofmt on these files?

senanz replied:
Same as the above comment the format at the editor looks fine.

vaikas replied:
The editor might display things with spaces vs. tabs, but gofmt which ensures that regardless of which ide you use the code looks the same is not agreeing, for example:
https://github.com/sigstore/policy-controller/actions/runs/12412408356/job/34695099927?pr=1687

Did you try running gofmt? At the root if you do:
```
gofmt -s -w .
```

And then it should update everything for you.

senanz replied:
Done

vaikas commented on .github/workflows/kind-cluster-custom-resources.yaml:
```diff
@@ -0,0 +1,148 @@
+      fail-fast: false # Keep running if one leg fails.
+      matrix:
+        k8s-version:
+          - v1.27.x
```
@cpanato is updating the tested k8s versions, should proabably bump these once it merges.

senanz replied:
@cpanato - Could you please let me know when you merge your changes so i can make the right adjcments here?

vaikas replied:
This was done, so go ahead and adjust this matrix.

senanz replied:
Updated to:
- v1.29.x
- v1.30.x
- v1.31.x
- v1.32.x

senanz committed:
Merge branch 'sigstore:main' into issues_1388

Commit:
Merge branch 'issues_1388' of https://github.com/senanz/policy-controller into issues_1388

senanz committed:
Merge branch 'sigstore:main' into issues_1388

senanz committed:
Merge branch 'sigstore:main' into issues_1388

Commit:
Merge branch 'issues_1388' of https://github.com/senanz/policy-controller into issues_1388

senanz committed:
Merge branch 'sigstore:main' into issues_1388

senanz committed:
Merge branch 'sigstore:main' into issues_1388

Commit:
Merge branch 'issues_1388' of https://github.com/senanz/policy-controller into issues_1388

# Conflicts:
#	cmd/webhook/main.go

Commit:
Fix the formatting of the files and change misleading test name

vaikas commented on .github/workflows/kind-cluster-custom-resources.yaml:
```diff
@@ -141,6 +141,7 @@ jobs:
       - name: Run Custom resources Tests
         timeout-minutes: 5
         run: |
+          chmod +x ./test/e2e_test_policy_custom_resource.sh
```
FYI, you should be able to make this change in git so that you don't have to do this here.

senanz commented:
hi @0xiso , Will back to work on that early next week, will ping you on slack and for sure we can collaborate.

Cajga commented:
Hi @senanz , sorry for bothering but do you still plan to give this a go?

hectorj2f commented on test/e2e_test_policy_custom_resource.sh:
```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env bash
+#
+# Copyright 2024 The Sigstore Authors.
```
2025
